### PR TITLE
Draft: Do fresh build each time

### PR DESCRIPTION
**Pull request by orhtej2**
## Problem

https://paste.yunohost.org/raw/wuviduxote

## Solution

Nuke the entire site from orbit.

## PR Status

- [x] **DRAFT DO NOT MERGE** (without proper testing, seems to work just fine)
- [ ] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)
 - Check if all runtime artifacts are stored in `$data_dir`
 - Check if config is perserved in `$data_dir`
 
## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by Tagadda** [comment]
!testme

File: scripts/upgrade [hunk 1]
```diff
@@ -33,7 +33,7 @@ then
 	ynh_script_progression --message="Upgrading source files..." --weight=2
 
 	# Download, check integrity, uncompress and patch the source from app.src
-	ynh_setup_source --dest_dir=$install_dir
+	ynh_setup_source --dest_dir=$install_dir -r
```
Comment (Tagadda) [inline scripts/upgrade]:
```suggestion
	ynh_setup_source --dest_dir=$install_dir --full_replace=1
```

**Comment by orhtej2** [comment]
I'm not sure if it does not delete anything important. Doesn't seem so but never tested.

Also if we're to nuke the install dir every time then perhaps it's a good idea to clean `yarn` caches.

**Comment by orhtej2** [comment]
Stanza from `element-web` build:

```
ynh_secure_remove --file="$install_dir/.cache"
ynh_secure_remove --file="$install_dir/.yarn"
ynh_secure_remove --file="$install_dir/.config"
ynh_secure_remove --file="$install_dir/.npm"
```

Note I never actually installed this app, one should check what's inside after the build.

**Comment by orhtej2** [comment]
Apparently solves #61

**Comment by orhtej2** [comment]
!testme

**Pull request merged**